decode/clarity.ts

```diff
@@ -1,40 +1,50 @@
 import version from "../src/core/version";
-import { Event, IAugmentation, IDecodedEvent, IDecodedPayload, IPayload, Token } from "../types/data";
-import data from "./data";
-import diagnostic from "./diagnostic";
-import envelope from "./envelope";
-import interaction from "./interaction";
+import { Augmentation, Event, Payload, Token } from "../types/data";
+import * as Decode from "../types/decode";
+import * as data from "./data";
+import * as diagnostic from "./diagnostic";
+import * as interaction from "./interaction";
 import * as layout from "./layout";
-import metric from "./metric";
 import * as r from "./render";
-import * as summary from "./summary";
 
 let pageId: string = null;
 
-export function decode(input: string | IPayload, augmentations: IAugmentation = null): IDecodedPayload {
-    let json: IPayload = typeof input === "string" ? JSON.parse(input) : input;
-    let timestamp = augmentations ? augmentations.timestamp : Date.now();
-    let ua = augmentations ? augmentations.ua : (navigator && "userAgent" in navigator ? navigator.userAgent : "");
-    let payload: IDecodedPayload = { timestamp, ua, envelope: envelope(json.e), metrics: metric(json.m), analytics: [], playback: [] };
+export function decode(input: string | Payload, augmentations: Augmentation = null): Decode.DecodedPayload {
+    let json: Payload = typeof input === "string" ? JSON.parse(input) : input;
+    let envelope = data.envelope(json.e);
+    let timestamp = augmentations && augmentations.timestamp ? augmentations.timestamp : Date.now();
+    let ua = augmentations && augmentations.ua ? augmentations.ua : (navigator && "userAgent" in navigator ? navigator.userAgent : "");
+    let payload: Decode.DecodedPayload = { timestamp, envelope };
+    if (envelope.sequence === 1) { payload.ua = ua; }
     let encoded: Token[][] = json.d;
 
     if (payload.envelope.version !== version) {
         throw new Error(`Invalid Clarity Version. Actual: ${payload.envelope.version} | Expected: ${version}`);
     }
 
     /* Reset components before decoding to keep them stateless */
-    summary.reset();
+    data.reset();
     layout.reset();
 
     for (let entry of encoded) {
-        let event: IDecodedEvent;
-        summary.decode(entry);
+        data.summarize(entry);
         switch (entry[1]) {
-            case Event.Scroll:
-            case Event.Document:
-            case Event.Resize:
-            case Event.Selection:
-            case Event.Change:
+            case Event.Page:
+                if (payload.page === undefined) { payload.page = []; }
+                payload.page.push(data.decode(entry) as Decode.PageEvent);
```

Comment:
>page [](start = 24, length = 4)

Can we change these so that we have a single pointer event and then whether touch or not touch is just a property. Makes handling these a bit clearer as the touch vs. non-touch isn't that central to handling them. So the y would be Pointer{Up,Down,Move,Cancel} for the corresponding events.

Reply:
There's not a lot of overlap between mouse & touch events:
- MouseUp | MouseDown | MouseMove | MouseWheel
- TouchStart | TouchCancel | TouchMove | TouchEnd

So the only thing that is common is *Move. Let's discuss this offline, won't fix for now.

Reply:
Actually there is as shown by the pointer event enent model where MouseUp <-> TouchEnd, MouseDown <-> TouchStart MouseMove<->TouchMove, MouseCacnel/PointerCancle<->TouchCancel PointerError/
Agree, we can refine this offline for next iteration. 

---
In reply to: [329382328](https://github.com/microsoft/clarity-js/pull/149#discussion_r329382328) [](ancestors = 329382328)